pkg/probe/checker.go

```diff
@@ -46,9 +47,18 @@ func (c *Checker) GetAllPeerPods() (ready bool, err error) {
 	for _, pod := range pods.Items {
 		if pod.Spec.RuntimeClassName != nil && *pod.Spec.RuntimeClassName == c.RuntimeclassName {
 			// peer-pods
-			logger.Printf("Dealing with PeerPod: %s, in phase: %s", pod.ObjectMeta.Name, pod.Status.Phase)
-			if pod.Status.Phase != corev1.PodRunning {
-				return false, fmt.Errorf("PeerPod %s is in %s phase.", pod.ObjectMeta.Name, pod.Status.Phase)
+			for _, condition := range pod.Status.Conditions {
+				if condition.Type == corev1.PodReady {
+					logger.Printf("Dealing with PeerPod: %s, with Ready condition: %v", pod.ObjectMeta.Name, condition)
+					if condition.Status != corev1.ConditionTrue {
+						return false, fmt.Errorf("PeerPod %s is not Ready.", pod.ObjectMeta.Name)
+					}
+
+					lastTransitionTime := condition.LastTransitionTime.Time
```

Comment:
Sounds L57 and 58 can be merged? Like:
`if condition.LastTransitionTime.Time.Before(startTime)`

Reply:
OK, that would be better.